File: MATLAB/Conversions to C/Softmax/Softmax.cpp


// Include files
#include "Softmax.h"
#include <cmath>

// Function Definitions
void Softmax(const double x[10], double y[10])
{
  double b_y;
  for (int k{0}; k < 10; k++) {
    y[k] = std::exp(x[k]);
  }
  b_y = y[0];
  for (int k{0}; k < 9; k++) {
    b_y += y[k + 1];
  }
  for (int k{0}; k < 10; k++) {
    y[k] /= b_y;
  }
}

// End of code generation (Softmax.cpp)

File: MATLAB/CNN/CNN_FloatingPoint_IntermediateSteps_TestBench/Softmax.m
function y = Softmax(x)
%ex = exp(x);
%ex = taylor_exp(x,90);
%y = ex / sum(ex);

ex = zeros(size(x));
for i=1:length(x)
     ex(i) = exp(x(i));
end 
y = ex / sum(ex);
end